File: TAM.LogisticSystem/client/js/Services/MasterRitasePriceService.ts



export class MasterRitasePriceService {
    static $inject = ['$http'];
    constructor(httpService: angular.IHttpService) {
        this.httpService = httpService;
    }
    

    //get All Ritase Data
    getAllRitaseData() {
        return this.httpService.get<MasterRitasePriceViewModel[]>('/api/v1/MasterRitasePriceApi/');
    }

    //delete Ritase Data by id
    deleteData(id: number) {
        return this.httpService.delete('/api/v1/MasterRitasePriceApi/DeleteData/'+id);
    }

    //get all delivery vendor code
    getDeliveryVendor() {
        return this.httpService.get<DeliveryVendor[]>('/api/v1/MasterRitasePriceApi/GetDeliveryVendor');
    }


    //get all delivery method code
    getDeliveryMethod() {
        return this.httpService.get<DeliveryMethod[]>('/api/v1/MasterRitasePriceApi/GetDeliveryMethod');
    }

    //get all city leg code
    getCityLeg() {
        return this.httpService.get<CityLeg[]>('/api/v1/MasterRitasePriceApi/GetCityLegCode');
    }

    getCurrencySymbol() {
        return this.httpService.get<CurrencySymbol[]>('/api/v1/MasterRitasePriceApi/GetCurrencySymbol');
    }


    //add ritase data
    addData(data: MasterRitasePriceInput) {
        return this.httpService.post('/api/v1/MasterRitasePriceApi/AddData', data);
    }

    //update data
    updateData(data: MasterRitasePriceViewModel) {
        return this.httpService.put('/api/v1/MasterRitasePriceApi/UpdateData', data);
    }

    //property
    httpService: angular.IHttpService;
}

export class TryModell {
    Check: string;
    Numb: number;
}

export class MasterRitasePriceInput {
    cityLegCode: string;
    currencySymbol: string;
    deliveryMethodCode: string;
    deliveryVendorCode: string;
    isSingleTrip: boolean;
    nominal: number;
    validDate: Date;
}

export class DeliveryVendor {
    deliveryVendorCode: string;
}

export class CityLeg {
    cityLegCode: string;
}

export class DeliveryMethod {
    deliveryMethodCode: string;
}

export class CurrencySymbol {
    currencySymbol: string;
}


export class MasterRitasePriceViewModel {
    cityLegCode: string;
    cityLegRitaseCostId: number;
    currencySymbol: string;
    deliveryMethodCode: string;
    deliveryVendorCode: string;
    isSingleTrip: string;
    nominal: number;
    validDate: Date;
}

export class Search {
    cityLegCode: string;
    cityLegRitaseCostId: number;
    currencySymbol: string;
    deliveryMethodCode: string;
    deliveryVendorCode: string;
    isSingleTrip: string;
    nominal: number;
    validDate: Date;
}

export class TripList {
    code: number;
    name: string;

}

File: TAM.LogisticSystem/Services/MasterRitasePriceService.cs
using Dapper;
using Microsoft.EntityFrameworkCore;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using TAM.LogisticSystem.Entities;
using TAM.LogisticSystem.Models;

namespace TAM.LogisticSystem.Services
{
    public class MasterRitasePriceService
    {

        public MasterRitasePriceService( LogisticDbContext logisticDbContext)
        {
            _TangoDbContext = logisticDbContext;
        }
        private readonly LogisticDbContext _TangoDbContext;

        // TIE: START
        ///// <summary>
        ///// Delete Ritase data by id
        ///// </summary>
        ///// <param name="id"></param>
        ///// <returns></returns>
        //public async Task DeleteData(int id)
        //{
        //    var selected = await _TangoDbContext.CityLegRitaseCost.AsNoTracking()
        //                    .FirstOrDefaultAsync(q => q.CityLegRitaseCostId == id);

        //    _TangoDbContext.Remove(selected);
        //    await _TangoDbContext.SaveChangesAsync();
        //}

        ///// <summary>
        ///// Get all Delivery vendor code
        ///// </summary>
        ///// <returns></returns>
        //public async Task<List<MasterRitasePriceDeliveryVendor>> GetDeliveryVendorCode()
        //{
        //    var data = await _TangoDbContext.DeliveryVendor.AsNoTracking().Select(q =>
        //    new MasterRitasePriceDeliveryVendor
        //    {
        //        DeliveryVendorCode = q.DeliveryVendorCode
        //    }).ToListAsync();

        //    return data;
        //}

        ///// <summary>
        ///// Get All Delivery Method code
        ///// </summary>
        ///// <returns></returns>
        //public async Task<List<MasterRitasePriceDeliveryMethod>> GetDeliveryMethodCode()
        //{
        //    var data = await _TangoDbContext.DeliveryMethod.AsNoTracking().Select(q =>
        //    new MasterRitasePriceDeliveryMethod
        //    {
        //        DeliveryMethodCode = q.DeliveryMethodCode
        //    }).ToListAsync();

        //    return data;
        //}


        ///// <summary>
        ///// Get All City Leg code
        ///// </summary>
        ///// <returns></returns>
        //public async Task<List<MasterRitasePriceCityLeg>> GetCityLegCode()
        //{
        //    var data = await _TangoDbContext.CityLeg.AsNoTracking().Select(q =>
        //    new MasterRitasePriceCityLeg
        //    {
        //        CityLegCode = q.CityLegCode
        //    }).ToListAsync();

        //    return data;
        //}

        ///// <summary>
        ///// Get all Currency Symbol
        ///// </summary>
        ///// <returns></returns>
        //public async Task<List<MasterRitasePriceCurrencySymbol>> GetCurrencySymbol()
        //{
        //    var data = await _TangoDbContext.Currency.AsNoTracking().Select(q =>
        //    new MasterRitasePriceCurrencySymbol
        //    {
        //         CurrencySymbol= q.CurrencySymbol
        //    }).ToListAsync();

        //    return data;
        //}

        ///// <summary>
        ///// Add data Ritase
        ///// </summary>
        ///// <param name="data"></param>
        ///// <returns></returns>
        //public async Task AddData(MasterRitasePriceInputModel data)
        //{
        //    var newData = new CityLegRitaseCost
        //    {
        //        CityLegCode = data.CityLegCode,
        //        CurrencySymbol = data.CurrencySymbol,
        //        DeliveryMethodCode = data.DeliveryMethodCode,
        //        DeliveryVendorCode = data.DeliveryVendorCode,
        //        IsSingleTrip = data.IsSingleTrip,
        //        Nominal = data.Nominal,
        //        ValidDate = data.ValidDate
        //    };
        //    _TangoDbContext.CityLegRitaseCost.Add(newData);
        //    await _TangoDbContext.SaveChangesAsync();
        //}

        ////Update Data 
        //public async Task UpdateData( MasterRitasePriceEditModel data)
        //{
        //    var dataSelected = await _TangoDbContext.CityLegRitaseCost.FirstOrDefaultAsync(q => q.CityLegRitaseCostId == data.CityLegRitaseCostId);
        //    dataSelected.CityLegCode = data.CityLegCode;
        //    dataSelected.CurrencySymbol = data.CurrencySymbol;
        //    dataSelected.DeliveryMethodCode = data.DeliveryMethodCode;
        //    dataSelected.DeliveryVendorCode = data.DeliveryVendorCode;
        //    if(data.IsSingleTrip=="Single Trip")
        //    {
        //        dataSelected.IsSingleTrip = false;
        //    } else
        //    {
        //        dataSelected.IsSingleTrip = true;
        //    }
        //    dataSelected.ValidDate = data.ValidDate;
        //    dataSelected.Nominal = data.Nominal;

        //    _TangoDbContext.CityLegRitaseCost.Update(dataSelected);
        //    await _TangoDbContext.SaveChangesAsync();

        //}


        ////Get all Ritase data
        //public async Task<List<MasterRitasePriceViewModel>> GetAllRitaseData()
        //{
        //    var ritaseData = await _TangoDbContext.CityLegRitaseCost.AsNoTracking().Select(q => new MasterRitasePriceViewModel
        //    { CityLegCode = q.CityLegCode,
        //    CityLegRitaseCostId = q.CityLegRitaseCostId,
        //    CurrencySymbol = q.CurrencySymbol,
        //    DeliveryMethodCode=q.DeliveryMethodCode,
        //    DeliveryVendorCode=q.DeliveryVendorCode,
        //    IsSingleTrip = q.IsSingleTrip,
        //    Nominal= q.Nominal,
        //    ValidDate = q.ValidDate})
        //    .ToListAsync();
        //    return ritaseData;
        //}
        // TIE: END
    }
}
